app/projectsmodel.cpp

```diff
@@ -33,20 +33,49 @@ ProjectModel::~ProjectModel() {}
 
 void ProjectModel::findProjectFiles()
 {
+  /*
   QStringList entryList = QDir( mDataDir ).entryList( QDir::NoDotAndDotDot | QDir::Dirs );
   for ( QString folderName : entryList )
   {
     addProjectFromPath( mDataDir + "/" + folderName );
   }
+  */
+
+  // populate from mLocalProjects
+  mProjectFiles.clear();
+  const QList<LocalProjectInfo> projects = mLocalProjects.projects();
+  for ( const LocalProjectInfo &project : projects )
+  {
+    QDir dir( project.projectDir );
+    QFileInfo fi( project.qgisProjectFilePath );
+
+    ProjectFile projectFile;
+    projectFile.name = fi.fileName().remove( ".qgs" ).remove( ".qgz" );   // TODO: shouldn't this be mergin project name??
```

Comment:
Same as in Mergin, a project folder name is a name of the project. A real project filename isn't used.